src/editors/typescript/thingworxTypescriptGenerator.ts

```diff
@@ -165,7 +165,14 @@ export class ThingworxToTypescriptGenerator {
             }
             // now generate the service definition, as well as jsdocs
             classDefinition += this.generateJDocWithContent(serviceJsDocElements.join("\n"))
-            classDefinition += `${service.name} (params: ${entityName}.${service.name}Params): ${this.getTypescriptBaseType(outputMetadata)};\n`;
+            if(serviceParamList.length > 0) {
```

Comment:
Does this also cover the case where the service does have parameters defined but they are all optional? In that case the parameters object should ideally also be optional.